File: app/legacy-config-fields.js
/**
 * Legacy config field readers for profiles generated before English localization.
 * Legacy key names are centralized here so the rest of the codebase stays English-only.
 */
(function (root) {
  const LEGACY_NOTE_KEYS = [
    'note',
    'logic_cn',
    'keyword_cn',
    'query_cn',
    'keyword_zh',
    'query_zh',
    'zh',
  ];
  const LEGACY_TITLE_ALT_KEYS = ['title_alt', 'title_zh'];

  function readFirstString(obj, keys) {
    if (!obj || typeof obj !== 'object') return '';
    for (const key of keys) {
      const val = String(obj[key] || '').trim();
      if (val) return val;
    }
    return '';
  }

  function readNote(item) {
    return readFirstString(item, LEGACY_NOTE_KEYS);
  }

  function readTitleAlt(meta) {
    return readFirstString(meta, LEGACY_TITLE_ALT_KEYS);
  }

  root.LegacyConfigFields = {
    readNote,
    readTitleAlt,
  };
})(typeof window !== 'undefined' ? window : globalThis);

File: src/legacy_config_fields.py
"""Legacy config field readers for profiles generated before English localization.

Legacy key names are centralized here so the rest of the codebase stays English-only.
"""

from __future__ import annotations

from typing import Any, Mapping

LEGACY_NOTE_KEYS = (
    "note",
    "logic_cn",
    "keyword_cn",
    "query_cn",
    "keyword_zh",
    "query_zh",
    "zh",
)


def read_note(item: Any) -> str:
    if not isinstance(item, Mapping):
        return ""
    for key in LEGACY_NOTE_KEYS:
        value = item.get(key)
        if value is None:
            continue
        text = str(value).strip()
        if text:
            return text
    return ""
